src/vs/workbench/api/node/extHostTerminalService.ts

```diff
@@ -286,6 +286,7 @@ export class ExtHostTerminalService implements ExtHostTerminalServiceShape {
 	private _terminalProcesses: { [id: number]: ITerminalChildProcess } = {};
 	private _terminalRenderers: ExtHostTerminalRenderer[] = [];
 	private _getTerminalPromises: { [id: number]: Promise<ExtHostTerminal> } = {};
+	private _variableResolver: ExtHostVariableResolverService | undefined;
```

Comment:
This could be a `{ [workspaceRoot: string]: ExtHostVariableResolverService }`

Reply:
Actually I think we're meant to rebuild this._variableResolver whenever a folder is added or removed from the workspace?

Reply:
So I should register that on the onWorkspaceChange listener and calculate it then.